// @electron-forge/publisher-github/dist/util/github.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = void 0;
var _debug = _interopRequireDefault(require("debug"));
var _rest = require("@octokit/rest");
var _pluginRetry = require("@octokit/plugin-retry");
function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const logInfo = (0, _debug).default('electron-forge:publisher:github:info');
const logDebug = (0, _debug).default('electron-forge:publisher:github:debug');
class GitHub {
    constructor(authToken = undefined, requireAuth = false, options = {}){
        const noOp = ()=>{
        /* Intentionally does nothing */ };
        this.options = {
            ...options,
            log: {
                debug: logDebug.enabled ? logDebug : noOp,
                error: console.error,
                info: logInfo.enabled ? logInfo : noOp,
                warn: console.warn
            },
            userAgent: 'Electron Forge'
        };
        if (authToken) {
            this.token = authToken;
        } else if (process.env.GITHUB_TOKEN) {
            this.token = process.env.GITHUB_TOKEN;
        } else if (requireAuth) {
            throw new Error('Please set GITHUB_TOKEN in your environment to access these features');
        }
    }
    getGitHub() {
        const options = {
            ...this.options
        };
        if (this.token) {
            options.auth = this.token;
        }
        const RetryableOctokit = _rest.Octokit.plugin(_pluginRetry.retry);
        const github = new RetryableOctokit(options);
        return github;
    }
}
exports.default = GitHub;
